client/components/domains/register-domain-step/index.jsx

```diff
@@ -584,20 +583,36 @@ const RegisterDomainStep = React.createClass( {
 				}
 				break;
 
-			case 'mappable_but_forbidden_subdomain':
+			case 'not_mappable_forbidden_subdomain':
 				message = this.translate( 'Subdomains starting with \'www.\' cannot be mapped to a WordPress.com blog' );
 				break;
 
-			case 'mappable_but_mapped_domain':
+			case 'not_mappable_invalid_tld':
+			case 'invalid_query':
+				message = this.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
+					args: { domain: domain }
+				} );
+				break;
+
+			case 'not_mappable_mapped_domain':
 				message = this.translate( 'This domain is already mapped to a WordPress.com site.' );
 				break;
 
-			case 'mappable_but_restricted_domain':
+			case 'not_mappable_restricted_domain':
```

Comment:
That's how it comes from the API and yes, it means always that it's a WordPress subdomain. Eh, I knew I shouldn't have touched anything else and left this mess as is - I have my reward 😛 But I'm not going to change the API - that's way beyond the scope of this PR. Especially since renaming this error code is not that important. This client (Calypso) knows how to interpret it and any other client will figure it out based on the full error message:
`'restricted_domain', 'WordPress.com domains cannot be mapped to a WordPress.com blog'`.